lib/ipmi_backend_utils.pm

```diff
@@ -213,6 +213,42 @@ sub get_installation_partition {
     return $partition;
 }
 
+sub set_dom0_mem {
+    my ($root_prefix) = @_;
+    $root_prefix = "/" if (!defined $root_prefix) || ($root_prefix eq "");
+    my $installation_disk = "";
+
+    if ($root_prefix ne "/") {
+        $installation_disk = get_installation_partition;
+        assert_script_run("cd /");
+        mount_installation_disk("$installation_disk", "$root_prefix");
+    }
+
+    assert_script_run('mount --rbind /proc /mnt/proc');
+    assert_script_run('mount --rbind /sys /mnt/sys');
+    assert_script_run('mount --rbind /dev /mnt/dev');
+    type_string("chroot /mnt\n");
+    wait_still_screen;
+
+    # Mount Btrfs sub-volumes
+    assert_script_run('mount -a');
+
+    assert_script_run ". /etc/default/grub";
+    assert_script_run "sed -i '/GRUB_CMDLINE_XEN_DEFAULT/c\\GRUB_CMDLINE_XEN_DEFAULT=\"\$GRUB_CMDLINE_XEN_DEFAULT dom0_mem=4096M\"' /etc/default/grub";
```

Comment:
Suggest to make this dom0_mem configurable.

Reply:
```suggestion
my $xen_dom0_mem = get_var('XEN_DOM0_MEM', '4096M');
assert_script_run "sed -i '/GRUB_CMDLINE_XEN_DEFAULT/c\\GRUB_CMDLINE_XEN_DEFAULT=\"\$GRUB_CMDLINE_XEN_DEFAULT dom0_mem=$xen_dom0_mem\"' /etc/default/grub";
```
is this OK?